### PR TITLE
virttest.utils_net: refactor function update_mac_ip_address

### PR DESCRIPTION
**Pull request by xutian**
It is unreasonable to update MAC address of the virtnet according to address in guest OS one by one. NIC index if virtnet not always equal NIC index in guest OS. This commit skip update virtnet, only update VM address_cache.

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3196,45 +3197,47 @@ def get_linux_ipaddr(session, nic):
...
+        line = re.sub(r"[\"\'\{\}]", "", line)
+        addr_info = map(lambda x: x.strip(), line.split(","))
+        mac = addr_info.pop().lower().replace("-", ":")
+        addrs = filter(None, map(str2ipaddr, addr_info))
+        ipv4_addr = filter(lambda x: x.version == 4, addrs)
+        ipv6_addr = filter(lambda x: x.version == 6, addrs)
+        if ipv4_addr:
+            maps[mac] = str(ipv4_addr[0])
+        if ipv6_addr:
+            maps["%s_6" % mac] = str(ipv6_addr[0])
```
Comment (luckyh) [inline virttest/utils_net.py]: I suggest to use the following structure to represent it.
```python
maps = {
    '52:54:00:24:52:a1': {
        'inet': '192.168.122.172',
        'inet6': (
            'fe80::6070:31b0:523e:cf5f',
            ...
        )
    },
    ...
}
```

File: virttest/utils_net.py [hunk 2]
```diff
@@ -3245,11 +3248,16 @@ def linux_mac_ip_maps(session):
     maps = {}
     for nic in get_linux_ifname(session):
         mac = get_linux_mac(session, nic)
-        maps[mac] = get_linux_ipaddr(session, nic)
+        mac = mac.lower().replace("-", ":")
+        ipv4_addr, ipv6_addr = get_linux_ipaddr(session, nic)
+        if ipv4_addr:
+            maps[mac] = ipv4_addr
+        if ipv6_addr:
+            maps["%s_6" % mac] = ipv6_addr
```
Comment (luckyh) [inline virttest/utils_net.py]: Suggest the same structure as the windows one.

File: virttest/utils_net.py [hunk 3]
```diff
@@ -3196,45 +3197,47 @@ def get_linux_ipaddr(session, nic):
...
-            inet = None
-        try:
-            inet6 = inets[1]
-        except IndexError:
-            inet6 = None
-        maps[mac] = (inet, inet6)
-
+        line = re.sub(r"[\"\'\{\}]", "", line)
+        addr_info = map(lambda x: x.strip(), line.split(","))
+        mac = addr_info.pop().lower().replace("-", ":")
```
Comment (luckyh) [inline virttest/utils_net.py]: I'm afraid the above codes can not correctly get the mac address, at least for win7.
```powershell
C:\> wmic nicconfig where IPEnabled=True get ipaddress, macaddress
IPAddress                                         MACAddress         
{"192.168.122.172", "fe80::6070:31b0:523e:cf5f"}  52:54:00:24:52:A1
```

File: virttest/utils_net.py [hunk 4]
```diff
@@ -3317,32 +3325,16 @@ def update_mac_ip_address(vm, params, timeout=None):
...
-                                                   "MACAddress",
-                                                   mac,
-                                                   "NetConnectionID")
-            if ifname:
-                logging.warn("'%s' not belong to VM '%s'" % (ifname, vm.name))
-        vm.address_cache[mac] = maps[key][0]
-        vm.address_cache["%s_6" % mac] = maps[key][1]
+        session = vm.wait_for_serial_login(timeout=timeout,
+                                           restart_network=True)
+    addr_map = get_guest_address_map(session)
```
Comment (luckyh) [inline virttest/utils_net.py]: Better to move this line into the if section for avoid read overhead.

Comment (luckyh) [inline virttest/utils_net.py]: And the below `session.close()` line.

File: virttest/utils_net.py [hunk 5]
```diff
@@ -3196,45 +3197,47 @@ def get_linux_ipaddr(session, nic):
     """
     Get IP addresses by nic name
     """
+    rex = r"inet6?\s+(\S+)\s+"
```
Comment (xutian) [inline virttest/utils_net.py]: thanks your comments,  will fix in next version

File: virttest/utils_net.py (hunk 1 above)
Comment (xutian) [inline virttest/utils_net.py]: Yes,  above structure looks more beautiful but it need extra code to update vm.address_cache. keep current style,  I can use dict.update() update vm.address_cache directly.

File: virttest/utils_net.py (hunk 3 above)
Comment (xutian) [inline virttest/utils_net.py]: really a issue, thanks your comments, will fix in next version

**Comment by xutian** [comment]
@luckyh  commit updated, any other comments for this commit ?

**Comment by luckyh** [comment]
LGTM

**Comment by suqinhuang** [comment]
LGTM

**Comment by xutian** [comment]
thanks @ldoktor  and @suqinhuang , merged.

**Pull request merged**